target/cheri-common/cheri-translate-utils.h

```diff
@@ -1061,7 +1067,7 @@ static inline void gen_move_cap_gp_select_gp(DisasContext *ctx, int dest_num,
     cheri_tcg_printf_verbose("cwccc", format, dest_num, value,
                              tcg_cond_string(cond), true_source_num,
                              false_source_num);
-
+    /* FIXME: Why do we require that the register is decompressed here? */
```

Comment:
Presumably because not doing so complicates the TCG-level tracking of compression state

Reply:
Well with this change it should no longer be required. @LawrenceEsswood is that correct?

Reply:
This, somewhat. The tracking would be perfectly happy to have a union of possible states for a conditional move if it were a better choice. AFAICR, I decided that any move should decompress. This was mostly because move was being used for basically every instruction that had a capability write (which are implemented as a move, and then an in-place operation). I thought that if we were going to move a register, it was a good idea to decompress before rather than after (as it was likely going to be decompressed anyway). That way, the same amount of work was being done but leaving more things decompressed. My thinking was, if a register is being moved around we will probably at some point use its bounds.

Reply:
I guarantee some code will break if we remove it though. Although the get_base / get_top etc family ensure decompression, they do so via a 'gen_ensure_cap_decompressed', which kills temps. If we ever want to get rid of gen_ensure_cap_decompressed in move, lots of callers will likely have to do a gen_ensure_cap_decompressed just before the move anyway, before any useful temps are created.

Reply:
> TCG uses it for optimisation purposes, to avoid redundant decompression, done at translate time

Yes but we could just copy the translation context capreg state from the other register?

Reply:
But you are right that this tcg state tracking makes a difference, I accidentally removed the `gen_lazy_cap_set_state(ctx, dest_num, CREG_FULLY_DECOMPRESSED);` below instead of updating it an immediately got some tcg_aborts due to dead temps.